packages/core/lib/testing/soliditytest.js

```diff
@@ -1,178 +1,172 @@
-      if (err) return callback(err);
-
-      const config = runner.config;
-      let solcVersion = config.compilers.solc.version;
-      if (solcVersion === "native") solcVersion = new Native().load().version();
-      if (!solcVersion) SafeSend = "NewSafeSend.sol";
-      else if (semver.lt(semver.coerce(solcVersion), "0.5.0"))
-        SafeSend = "OldSafeSend.sol";
-      else SafeSend = "NewSafeSend.sol";
-
-      compile.with_dependencies(
-        runner.config.with({
-          paths: [
-            path.join(__dirname, "Assert.sol"),
-            path.join(__dirname, "AssertAddress.sol"),
-            path.join(__dirname, "AssertAddressArray.sol"),
-            // path.join(__dirname, "AssertAddressPayableArray.sol"), only compatible w/ ^0.5.0
-            path.join(__dirname, "AssertBalance.sol"),
-            path.join(__dirname, "AssertBool.sol"),
-            path.join(__dirname, "AssertBytes32.sol"),
-            path.join(__dirname, "AssertBytes32Array.sol"),
-            path.join(__dirname, "AssertGeneral.sol"),
-            path.join(__dirname, "AssertInt.sol"),
-            path.join(__dirname, "AssertIntArray.sol"),
-            path.join(__dirname, "AssertString.sol"),
-            path.join(__dirname, "AssertUint.sol"),
-            path.join(__dirname, "AssertUintArray.sol"),
-            "truffle/DeployedAddresses.sol", // generated by deployed.js
-            path.join(__dirname, SafeSend)
-          ],
-          quiet: true
-        }),
-        (err, contracts) => {
-          if (err) return callback(err);
-
-          // Set network values.
-          Object.keys(contracts).forEach(name => {
-            contracts[name].network_id = config.network_id;
-            contracts[name].default_network = config.default_network;
-          });
+  async compileNewAbstractInterface(runner) {
+    debug("compiling");
+    const config = runner.config;
+    const solcVersion = config.compilers.solc.version;
+    if (solcVersion === "native") {
+      solcVersion = new Native().load().version();
+    }
+    if (!solcVersion) {
+      SafeSend = "NewSafeSend.sol";
+    } else if (semver.lt(semver.coerce(solcVersion), "0.5.0")) {
+      SafeSend = "OldSafeSend.sol";
+    } else {
+      SafeSend = "NewSafeSend.sol";
+    }
 
-          config.artifactor
-            .saveAll(contracts)
-            .then(() => {
-              callback();
-            })
-            .catch(callback);
-        }
-      );
-    });
+    //HACK: promisify seems to lose context, so we explicitly bind compile
+    const contracts = await util.promisify(
+      compile.with_dependencies.bind(compile)
+    )(
+      runner.config.with({
+        paths: [
+          path.join(__dirname, "Assert.sol"),
+          path.join(__dirname, "AssertAddress.sol"),
+          path.join(__dirname, "AssertAddressArray.sol"),
+          // path.join(__dirname, "AssertAddressPayableArray.sol"), only compatible w/ ^0.5.0
```

Comment:
Should we just delete this?

Reply:
We probably should!  I decided not to mess with the commented-out code (OK, I updated it in places :P ) but I didn't delete it because I wasn't sure what it was for.  If anyone who knows (@gnidan?) can chime in, it'd be helpful.

Reply:
delete it. it's in the git history if we want it again